components/eventing-controller/controllers/subscriptionv2/eventmesh/test/reconciler_integration_test.go

```diff
@@ -43,6 +46,71 @@ func TestMain(m *testing.M) {
 	os.Exit(code)
 }
 
+func Test_ValidationWebhook(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		name                  string
+		givenSubscriptionFunc func(namespace string) *eventingv1alpha2.Subscription
+		wantError             error
+	}{
+		{
+			name: "should fail to create subscription with invalid event source",
+			givenSubscriptionFunc: func(namespace string) *eventingv1alpha2.Subscription {
+				return reconcilertesting.NewSubscription(testName, namespace,
+					reconcilertesting.WithStandardTypeMatching(),
+					reconcilertesting.WithSource(""),
+					reconcilertesting.WithOrderCreatedV1Event(),
+					reconcilertesting.WithValidSink(namespace, "svc"),
+				)
+			},
+			wantError: generateInvalidSubscriptionError(testName, eventingv1alpha2.EmptyErrDetail, eventingv1alpha2.SourcePath),
+		},
+		{
+			name: "should fail to create subscription with invalid event types",
+			givenSubscriptionFunc: func(namespace string) *eventingv1alpha2.Subscription {
+				return reconcilertesting.NewSubscription(testName, namespace,
+					reconcilertesting.WithStandardTypeMatching(),
+					reconcilertesting.WithSource("source"),
+					reconcilertesting.WithTypes([]string{}),
+					reconcilertesting.WithValidSink(namespace, "svc"),
+				)
+			},
+			wantError: generateInvalidSubscriptionError(testName, eventingv1alpha2.EmptyErrDetail, eventingv1alpha2.TypesPath),
+		},
+		{
+			name: "should fail to create subscription with invalid event types",
```

Comment:
will change this in the next commit
```suggestion
			name: "should fail to create subscription with invalid sink",
```